porch/pkg/oci/mutate.go

```diff
@@ -201,7 +201,31 @@ func (p *ociPackageDraft) Close(ctx context.Context) (repository.PackageRevision
 
 	klog.Infof("pushing %s", ref)
 
-	img, err := mutate.Append(p.base, p.addendums...)
+	addendums := append([]mutate.Addendum{}, p.addendums...)
+	if p.lifecycle != "" {
+		if len(addendums) == 0 {
```

Comment:
When do we encounter this situation? Would this be when we approve an oci package, thus we want to update the lifecycle but there isn't any other changes to the content that requires a new layer?

Reply:
That's exactly right.  I think it will likely happen whenever we start approving packages.  I figure it's an important edge case that we're going to have to handle, but right now as I don't think we have test coverage I just surface an error.  I'd like to improve the test coverage here e.g. running all tests against both git & oci Repositories, as the API behaviour should be equivalent.